src/main/scala/uk/nationalarchives/dri/Lambda.scala

```diff
@@ -0,0 +1,117 @@
+package uk.gov.nationalarchives.dr2
+
+import cats.effect.IO
+import cats.effect.unsafe.implicits.global
+import com.amazonaws.services.lambda.runtime.events.ScheduledEvent
+import com.amazonaws.services.lambda.runtime.{Context, RequestHandler}
+import com.typesafe.config.{ConfigFactory, Config => TypeSafeConfig}
+import io.circe.Encoder
+import software.amazon.awssdk.services.dynamodb.model._
+import sttp.capabilities.fs2.Fs2Streams
+import uk.gov.nationalarchives.DADynamoDBClient.DynamoDbRequest
+import uk.gov.nationalarchives.dp.client.Entities.Entity
+import uk.gov.nationalarchives.dp.client.EntityClient
+import uk.gov.nationalarchives.dp.client.fs2.Fs2Client
+import uk.gov.nationalarchives.{DADynamoDBClient, DASNSClient}
+import uk.gov.nationalarchives.dr2.Lambda.CompactEntity
+
+import java.time.OffsetDateTime
+import java.util.{Date, TimeZone}
+
+class Lambda extends RequestHandler[ScheduledEvent, Unit] {
+  private val configFactory: TypeSafeConfig = ConfigFactory.load
+  private val apiUrl: String = configFactory.getString("api.url")
+  private val tableName: String = configFactory.getString("api.lastPolledTableName")
+  private val secretName: String = configFactory.getString("api.secretName")
+  private val snsArn: String = configFactory.getString("api.snsArn")
+  private val maxEntitiesPerPage: Int = 1000
+  private val dateItemPrimaryKeyAndValue =
+    Map("id" -> AttributeValue.builder().s("LastPolled").build())
+  private val datetimeField = "datetime"
+
+  def entitiesClientIO: IO[EntityClient[IO, Fs2Streams[IO]]] = Fs2Client.entityClient(apiUrl)
+
+  def dADynamoDBClient: DADynamoDBClient[IO] = DADynamoDBClient[IO]()
+
+  def dASnsDBClient: DASNSClient[IO] = DASNSClient[IO]()
```

Comment:
Can these three be `val` instead of `def`? `def` is re-evaluated every time you call it.